### PR TITLE
Add work wechat webhook

### PR DESCRIPTION
**Pull request by rocklau**
new feature: send events to work wechat group<chatid>

unit test done list
1. create new work wechat webhook
2. edit this webhook
3. push a commit, wechat group received this event

work flow: 

1. this command can get access token

get CORP_ID from https://work.weixin.qq.com/wework_admin/frame#profile

get APP_SECRET from your corporation app profile /wework_admin/frame#apps/modApiApp/

```bash
curl -X GET \
  'https://qyapi.weixin.qq.com/cgi-bin/gettoken?corpid=CORP_ID&corpsecret=CORP_APP_SECRET' 
```
```json 
{
    "errcode": 0,
    "errmsg": "ok",
    "access_token": "",
    "expires_in": 7200
}
```
2. and create your bot group 'gitea'
```bash
curl -X POST \
  'https://qyapi.weixin.qq.com/cgi-bin/appchat/create?access_token=ACCESS_TOKEN' \
  -H 'Content-Type: application/json' \
  -d '{
    "name" : "Gitea",
    "owner" : "XiaoMing",
    "userlist" : ["XiaoMing","XiaoHong"],
    "chatid" : "gitea"
}'
```
3. then we go to add new work wechat hook with access_token and chatid on gitea

TargetURL:https://qyapi.weixin.qq.com/cgi-bin/appchat/send?access_token=ACCESS_TOKEN

Chat ID: gitea

### CODE REVIEW

File: options/locale/locale_pt-BR.ini [hunk 1]
```diff
@@ -1115,6 +1115,7 @@ settings.slack_domain=Domínio
 settings.slack_channel=Canal
 settings.add_discord_hook_desc=Integre o <a href="%s">Discord</a> em seu repositório.
 settings.add_dingtalk_hook_desc=Integre o <a href="%s">Dingtalk</a> em seu repositório.
+settings.add_workwechat_hook_desc=Integre o <a href="%s">Work Wechat</a> em seu repositório.
```
Comment (techknowlogick) [inline options/locale/locale_pt-BR.ini]: Please add translations via https://crowdin.com/project/gitea

**Review by techknowlogick — CHANGES_REQUESTED** [review]

File: models/webhook_workwechat.go [hunk 2]
```diff
@@ -0,0 +1,405 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
```
Comment (lunny) [inline models/webhook_workwechat.go]: 2018

File: options/locale/locale_pt-BR.ini (hunk 1 above)
Comment (rocklau) [reply options/locale/locale_pt-BR.ini]: ok.

File: options/locale/locale_de-DE.ini [hunk 3]
```diff
@@ -1115,6 +1115,7 @@ settings.slack_domain=Domain
 settings.slack_channel=Kanal
 settings.add_discord_hook_desc=<a href="%s">Discord</a>-Integration zu deinem Repository hinzufügen.
 settings.add_dingtalk_hook_desc=<a href="%s">Dingtalk</a>-Integration zu deinem Repository hinzufügen.
```
Comment (lunny) [inline options/locale/locale_de-DE.ini]: don't translate them.

File: routers/api/v1/convert/convert.go [hunk 4]
```diff
@@ -151,7 +151,10 @@ func ToHook(repoLink string, w *models.Webhook) *api.Hook {
 		config["icon_url"] = s.IconURL
 		config["color"] = s.Color
 	}
-
+	if w.HookTaskType == models.WORKWECHAT {
```
Comment (lunny) [inline routers/api/v1/convert/convert.go]: else if

File: routers/repo/webhook.go [hunk 5]
```diff
@@ -106,6 +106,11 @@ func WebhooksNew(ctx *context.Context) {
 			"IconURL":  setting.AppURL + "img/favicon.png",
 		}
 	}
+	if hookType == "workwechat" {
```
Comment (lunny) [inline routers/repo/webhook.go]: else if

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
Please resolve conflicts.

**Comment by lunny** [comment]
Please resolve conflicted files

**Comment by 6543** [comment]
@rocklau do you still work on this :( ?

**Comment by lunny** [comment]
Please resolve the conflicts.

**Comment by 6543** [comment]
@rocklau #9422 made some changes ... so It is worth looking at it ... (you have to resolve conflicts anyway)

**Comment by 6543** [comment]
@rocklau  I know this PR is old ... but do you still have passion to get it upstream?

**Pull request closed without merging**